task6/controllers/api/loginUser.controller.ts

```diff
@@ -0,0 +1,48 @@
+import { Request, Response } from 'express';
+import { getUserByEmail, updateUser } from '../../repositories/users.repository';
+import { userBodySchema } from './registerUser.controller';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+
+export async function loginUser(req: Request, res: Response) {
+  const body = req.body;
+
+  const data = userBodySchema.validate(body);
+
+  if (data.error) {
+    res.status(400).send(`User data is not valid: ${data.error.message}`);
+    return;
+  }
+
+  const user = await getUserByEmail(data.value.email);
+
+  if (!user) {
+    res.status(400).send(`User with email '${data.value.email}' does not exist!`);
+    return;
+  }
+
+  const isPasswordMatches = bcrypt.compareSync(data.value.password, user.password);
+
+  if (!isPasswordMatches) {
+    res.status(403).send('Incorrect password');
```

Comment:
I think in this case should be 401 status code.
For testing examples sometimes used https://www.npmjs.com/package/http-status-codes
or create own implementation for this can help later